src/version.rs

```diff
@@ -300,16 +300,17 @@ pub fn dependencies(req: &mut Request) -> CargoResult<Response> {
 
 /// Handles the `GET /crates/:crate_id/:version/downloads` route.
 pub fn downloads(req: &mut Request) -> CargoResult<Response> {
+    let (offset, limit) = req.pagination(90, 100)?;
```

Comment:
`YYYY-MM-DD`. I'd just pass it directly as a bind param into the query if Rust postgres lets us. If not, `strptime` it up